lib/ace/ext/papeeria/katex-previewer.coffee

```diff
@@ -1,5 +1,106 @@
 define((require, exports, module) ->
   LatexParsingContext = require("ace/ext/papeeria/latex_parsing_context")
+  TokenIterator = require("ace/token_iterator").TokenIterator
+  Range = require("ace/range").Range
+
+  getEquationRangeHandler = (editor) ->
+    erh = {
+      BEGIN_EQUATION_TOKEN_SEQUENCE: [
+        {
+          type: "storage.type"
+          value: "\\begin"
+        }
+        {
+          type: "lparen"
+          value: "{"
+        }
+        {
+          type: "variable.parameter"
+          value: "equation"
+        }
+        {
+          type: "rparen"
+          value: "}"
+        }
+      ]
+      END_EQUATION_TOKEN_SEQUENCE: [
+        {
+          type: "storage.type"
+          value: "\\end"
+        }
+        {
+          type: "lparen"
+          value: "{"
+        }
+        {
+          type: "variable.parameter"
+          value: "equation"
+        }
+        {
+          type: "rparen"
+          value: "}"
+        }
+      ]
+
+      compareTokens: (token1, token2) ->
+        if token1? and token2?
+          return token1.type == token2.type and token1.value == token2.value
+        else return if token1? or token2? then false else true
+
+      getEquationStart: (tokenIterator) ->
+        # following cycle pushes tokenIterator to the end of
+        # beginning sequence, if it is inside one
+        for token in erh.BEGIN_EQUATION_TOKEN_SEQUENCE
+          if erh.compareTokens(token, tokenIterator.getCurrentToken())
+            tokenIterator.stepForward()
+        curSequenceIndex = erh.BEGIN_EQUATION_TOKEN_SEQUENCE.length - 1
+        curEquationStart = null
+        while curSequenceIndex >= 0
+          if erh.compareTokens(
+              erh.BEGIN_EQUATION_TOKEN_SEQUENCE[curSequenceIndex],
+              tokenIterator.stepBackward()
+          )
```

Comment:
any sense to surround the body of this if with curly brackets, just to make it explicit?